### ISSUE TITLE
content_changed implementation

### ISSUE THREAD

**Issue by rw8896** [DMTF/libspdm#2816]
https://github.com/DMTF/libspdm/blob/704d5652b2eeb451fffbd1bb1129046e377c9cf3/include/hal/library/responder/measlib.h#L77

Hi,

I have a question about content_changed implementation. 
The integrator will need the knowledge about measurement logs in current L2 contents as a basis to check if there is any measurement changed. But the function didn't seem to be able to provide such information.
Is there a design/mechanism for integrators to acquire the information from libspdm?

**Comment by steven-bellock** [comment]
The Integrator needs to monitor `request_attribute` and the `SignatureRequested` bit, since `content_change` can only be asserted when a signature is requested. So it keeps track of measurements when `SignatureRequested` is not set, and does a comparison when `SignatureRequested` is set.

**Comment by rw8896** [comment]
Thanks and I think that works for normal scenario. But as "Any communication between Requester and Responder other than a GET_MEASUREMENTS request or response re-initializes L1/L2 computation to null .", the integrator seems not able to know if L1/L2 has been re-inited?

**Comment by steven-bellock** [comment]
Yes, that looks like it's an issue. The state of the l1/l2 transcript is stored in the `spdm_context` but it is currently not public. We would need to provide a public function to query the transcript state. @jyao1 does that sound good?

**Comment by rw8896** [comment]
I feel a callback function to notify L1/L2 was reset might be easier for integrators to track the measurement.
Maybe provide a callback function for transcript state change (similar to connection state callback)?

**Comment by steven-bellock** [comment]
I will draw up the two approaches and we can see which ones looks the best: callback versus querying the transcript from the `spdm_context`.

**Comment by jyao1** [comment]
Some thought:

Internal L1/L2 states:
* Reset <= initial state
* Started <= tracking state
* Changed <= content_changed 

State machine:
* Reset -> Started : when first unsigned measurement is returned.
* Reset -> Reset : after signed measurement is returned.
* Started -> Started: when non-first unsigned measurement is returned.
* Started -> Changed : when firmware is updated for the returned index.
* Started -> Reset: when L1/L2 is reset to NUL.
* Changed -> Changed: when non-first unsigned measurement is return.
* Changed -> Reset: after signed measurement is returned, (or) when L1/L2 is reset to NUL.

**Comment by steven-bellock** [comment]
>Maybe 1), since it seems a required feature?

`Content changed` is an optional feature, so I would go with number 2. 

In addition, `libspdm_measurement_collection` needs to know the `session_id`.

**Comment by rw8896** [comment]
> `Content changed` is an optional feature, so I would go with number 2.
> 
Yes, the second approach seems more appropriate, as Steven pointed out that "Content changed" is an optional feature.

**Comment by rw8896** [comment]
Since the l1l2 is maintained per session, in addition to [add libspdm_meas_log_reset_callback_func](https://github.com/Li-Aaron/libspdm/commit/e4d715b1301dc5eaa7010e3fc28039e86c1f9023), does it also need to add session_info to libspdm_measurement_collection?
